crypto/rsa/rsa_impl.c

```diff
@@ -644,11 +635,10 @@ static int mod_exp(BIGNUM *r0, const BIGNUM *I, RSA *rsa, BN_CTX *ctx) {
     goto err;
   }
 
-  /* Turn BN_FLG_CONSTTIME flag on before division operation */
-  pr1 = &local_r1;
-  BN_with_flags(pr1, r1, BN_FLG_CONSTTIME);
+  /* Ensure BN_FLG_CONSTTIME flag is on before division operation */
+  assert(BN_get_flags(r1, BN_FLG_CONSTTIME));
```

Comment:
> does that mean this code isn't being run during the tests at all?

It looks like this code is being run in the tests. I put an `assert(0)` here and things seem to fail normally.

If I do this:

```c
  BN_CTX_start(ctx);
  r1 = BN_CTX_get(ctx);
  m1 = BN_CTX_get(ctx);
  vrfy = BN_CTX_get(ctx);
  assert(BN_get_flags(r1, BN_FLG_CONSTTIME));
  assert(BN_get_flags(m1, BN_FLG_CONSTTIME));
  assert(BN_get_flags(vrfy, BN_FLG_CONSTTIME));
```

No assertion fail, which makes me think that we are reusing old entries in the `BN_CTX_POOL` that already have the flags set. Maybe we should reset the flags in `BN_CTX_get`?